### PR TITLE
Tests for prometheus::daemon

### PR DESCRIPTION
**Pull request by sathieu**

### CODE REVIEW

File: spec/fixtures/files/daemon.debian [hunk 1]
```diff
@@ -0,0 +1,161 @@
+#! /bin/sh
```
Comment (wyardley) [inline spec/fixtures/files/daemon.debian]: To me, having the entire body of both init scripts seems a bit like overkill, maybe?

File: spec/defines/daemon_spec.rb [hunk 2]
```diff
@@ -0,0 +1,157 @@
+require 'spec_helper'
+
+describe 'prometheus::daemon' do
+  let :title do
```
Comment (wyardley) [inline spec/defines/daemon_spec.rb:4]: should this be `:title` or `:params` and then setting `:title` in params?

File: spec/fixtures/files/daemon.debian (hunk 1 above)
Comment (TheMeier) [reply spec/fixtures/files/daemon.debian]: whats the point of checking the complete content of the init scripts/unit file anyway?

Comment (wyardley) [reply spec/fixtures/files/daemon.debian]: Yeah, that's what I mean.

File: spec/defines/daemon_spec.rb (hunk 2 above)
Comment (sathieu) [reply spec/defines/daemon_spec.rb:4]: this is usually ```:title```

File: spec/fixtures/files/daemon.debian (hunk 1 above)
Comment (sathieu) [reply spec/fixtures/files/daemon.debian]: the point is that it is hard to ensure correct file syntax with all those new lines and backslash. But your are the reviewer, tell me what you want.

Comment (TheMeier) [reply spec/fixtures/files/daemon.debian]: @sathieu i don't see how this check helps. if you change the files shipped in the module you need to change the  files shipped in the fixtures. so basically you are duplicating content without any benefit added

Comment (sathieu) [reply spec/fixtures/files/daemon.debian]: I generally agree, but this kind of rspec helped for https://github.com/voxpupuli/puppet-prometheus/pull/48/files (by ensuring that those ```<%= @daemon_flags.join(" \\\n          ") %>``` or like were correct)

Comment (TheMeier) [reply spec/fixtures/files/daemon.debian]: how about testing it this way
```
          if ['debian-7-x86_64'].include?(os)
            # init_style = 'debian'

            it {
              is_expected.to contain_file('/etc/init.d/smurf_exporter').with(
                'mode'    => '0555',
                'owner'   => 'root',
                'group'   => 'root',
              )
              should contain_file('/etc/init.d/smurf_exporter').with_content(/DAEMON_ARGS=''\n/)
            }
```
also prometheus::daemon doesn't actually set $daemon_flags so the join in this context always evaluates to ''

**Comment by sathieu** [comment]
Here is the updated PR. No more using fixtures.

**Review by wyardley — APPROVED** [review]
Looks pretty good to me.

**Review by juniorsysadmin — APPROVED** [review]

**Pull request merged**